src/Controllers/Ops.php

```diff
@@ -101,40 +102,21 @@ protected function saveToFile($file, $data, $package_name = null)
         return true;
     }
 
-    protected function recSave($pre_format, $key, $code, $v)
+    private static function explodeTranlsationKey(string $nestedTranslationKey, string|null $value): array
     {
-        if (strpos($key, '.')) {
-            $assoc = [];
-            $exp   = explode('.', $key);
+        $doneSplitting = !Str::contains($nestedTranslationKey, '.');
 
-            // convert array dot to associative
-            while (!empty($exp)) {
-                $assoc = [array_pop($exp) => $assoc];
-            }
-
-            // set value to last key
-            Arr::set($assoc, $key, $v);
-
-            // get root key
-            $first_key = strtok($key, '.');
-
-            // hacky fix to avoid indexed nesting
-            $final = array_pop($assoc);
-            $a     = key($final);
-            $b     = current($final);
+        if ($doneSplitting) return [$nestedTranslationKey => $value];
```

Comment:
If there is no more dots that means we have reached the end of the line and can return the last part of the key ($nestedTranslationKey) and the value of this translation.